web_sdk/test/api_conform_test.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:analyzer/analyzer.dart';
+
+int main() {
+  // These files just contain imports to the part files;
+  final CompilationUnit uiUnit = parseDartFile('lib/ui/ui.dart',
+      parseFunctionBodies: false, suppressErrors: false);
+  final CompilationUnit webUnit = parseDartFile('lib/stub_ui/ui.dart',
+      parseFunctionBodies: false, suppressErrors: false);
+  final Map<String, ClassDeclaration> uiClasses = <String, ClassDeclaration>{};
+  final Map<String, ClassDeclaration> webClasses = <String, ClassDeclaration>{};
+
+  // Gather all public classes from each library. For now we are skiping
+  // other top level members.
+  _collectPublicClasses(uiUnit, uiClasses, 'lib/ui/');
+  _collectPublicClasses(webUnit, webClasses, 'lib/stub_ui/');
+
+  if (uiClasses.isEmpty || webClasses.isEmpty) {
+    print('Warning: did not resolve all Classes');
+  }
+
+  bool failed = false;
+  print('Checking ${uiClasses.length} public classes.');
+  for (String className in uiClasses.keys) {
+    final ClassDeclaration uiClass = uiClasses[className];
+    final ClassDeclaration webClass = webClasses[className];
+    // If the web class is missing there isn't much left to do here. Print a
+    // warning and move along.
+    if (webClass == null) {
+      failed = true;
+      print('Warning: io/dart:ui contained public class $className, but '
```

Comment:
Could all these warning be made more actionable? For example, we could call out that people can find io/dart:ui in 'lib/ui/' and the other one in 'lib/stub_ui/' to make it clear where things need to be added when this fails.

Reply:
The new error message still doesn't look very actionable to me... Should I interpret it as "Please copy the missing classes and functions from lib/ui/ui.dart to lib/stub_ui/ui.dart."?